src/sentry/dynamic_sampling/prioritise_transactions.py

```diff
@@ -0,0 +1,307 @@
+import logging
+import time
+from datetime import datetime, timedelta
+from typing import Iterator, List, Optional, Tuple, TypedDict, cast
+
+from snuba_sdk import (
+    AliasedExpression,
+    Column,
+    Condition,
+    Direction,
+    Entity,
+    Function,
+    Granularity,
+    LimitBy,
+    Op,
+    OrderBy,
+    Query,
+    Request,
+)
+
+from sentry import options
+from sentry.sentry_metrics import indexer
+from sentry.snuba.dataset import Dataset, EntityKey
+from sentry.snuba.metrics.naming_layer.mri import TransactionMRI
+from sentry.snuba.referrer import Referrer
+from sentry.utils.snuba import raw_snql_query
+
+logger = logging.getLogger(__name__)
+MAX_SECONDS = 60
+CHUNK_SIZE = 10000
+
+
+class ProjectTransactions(TypedDict, total=True):
+    project_id: int
+    org_id: int
+    transaction_counts: List[Tuple[str, int]]
+
+
+def get_orgs_with_project_counts(max_orgs: int, max_projects: int) -> Iterator[List[int]]:
+    """
+    Fetch organisations in batches.
+    A batch will return at max max_orgs elements
+    It will accumulate org ids in the list until either it accumulates max_orgs or the
+    number of projects in the already accumulated orgs is more than max_projects or there
+    are no more orgs
+    """
+    start_time = time.time()
+    metric_id = indexer.resolve_shared_org(str(TransactionMRI.COUNT_PER_ROOT_PROJECT.value))
+    offset = 0
+    load_rate = int(options.get("dynamic-sampling.prioritise_transactions.load_rate") * 100)
+    last_result: List[Tuple[int, int]] = []
+    while (time.time() - start_time) < MAX_SECONDS:
+        query = (
+            Query(
+                match=Entity(EntityKey.GenericOrgMetricsCounters.value),
+                select=[
+                    Function("uniq", [Column("project_id")], "num_projects"),
+                    Column("org_id"),
+                ],
+                groupby=[
+                    Column("org_id"),
+                ],
+                where=[
+                    Condition(Function("modulo", [Column("org_id"), 100]), Op.LT, load_rate),
```

Comment:
This is fine for now. When you decide to change this to 100% and are satisfied with the results, please remove this modulo clause altogether.

Reply:
Yes, thanks @nikhars, this is the plan.